profiles/models.py

```diff
@@ -99,6 +100,25 @@ class Profile(models.Model):
     mailing_address = models.TextField(blank=True, null=True)
     date_joined_micromasters = models.DateTimeField(blank=True, null=True, auto_now_add=True)
     linkedin = JSONField(blank=True, null=True)
+    student_id = models.IntegerField(blank=True, null=True, unique=True)
+
+    @transaction.atomic
+    def save(self, *args, **kwargs):
+        """need to handle setting the student_id number"""
+        if self.id is None or self.student_id is None:
+            max_id = Profile.objects.aggregate(Max('student_id'))
+            if max_id['student_id__max'] is None:
+                max_id = 1
+            else:
+                max_id = max_id['student_id__max']
+            while Profile.objects.filter(student_id=max_id).exists():
+                max_id += 1
```

Comment:
I made a change so that each time `max_id` is incremented we also decrement a counter that starts at 1000. if this reaches zero we let the transaction happen and raise an exception.

does that do what we want?